agent.go

```diff
@@ -192,7 +192,6 @@ func (c *controller) agentSetup() error {
 	if remoteAddr != "" {
 		if err := c.agentJoin(remoteAddr); err != nil {
 			logrus.Errorf("Error in agentJoin : %v", err)
-			return nil
```

Comment:
This is not returning the error on the first place (returning nil). Am not sure how this will fix the issue mentioned in the description. The only possible effect this has is the fact that `agentInitDone` is not reinitialized. Is that the issue that you are intending to fix ?